rasa/nlu/selectors/response_selector.py

```diff
@@ -480,7 +505,11 @@ def _prepare_layers(self) -> None:
         self.label_name = TEXT if self.config[SHARE_HIDDEN_LAYERS] else LABEL
 
         self._prepare_sequence_layers(self.text_name)
-        self._prepare_sequence_layers(self.label_name)
+
+        # When training on labels, don't create transformer layers for the label
+        self._prepare_sequence_layers(
```

Comment:
just to clarify: the incoming user message is still going through the transformer. the responses are just going through the transformer if we are using the text, but not if we are using the label, in that case it is just a ffnn and an embedding layer, correct?

Reply:
Yes, that's correct. Just the transformer layer is skipped if training on labels.